libcontainer/nsenter/nsexec.c

```diff
@@ -150,76 +146,60 @@ static uint8_t readint8(char *buf)
 
 static void update_process_idmap(char *pathfmt, int pid, char *map, int map_len)
 {
-	char    buf[PATH_MAX];
-	int	len;
-	int	fd;
-
-	len = snprintf(buf, sizeof(buf), pathfmt, pid);
-	if (len < 0) {
-		pr_perror("failed to construct '%s' for %d", pathfmt, pid);
-		exit(1);
-	}
+	int	fd, len;
+	char path[PATH_MAX];
 
-	fd = open(buf, O_RDWR);
-	if (fd == -1) {
-		pr_perror("failed to open %s", buf);
-		exit(1);
-	}
+	len = snprintf(path, PATH_MAX, pathfmt, pid);
+	if (len < 0)
+		bail("failed to construct '%s' for %d", pathfmt, pid);
+
+	fd = open(path, O_RDWR);
+	if (fd < 0)
+		bail("failed to open %s", path);
 
 	len = write(fd, map, map_len);
-	if (len == -1) {
-		pr_perror("failed to write to %s", buf);
-		close(fd);
-		exit(1);
-	} else if (len != map_len) {
-		pr_perror("Failed to write data to %s (%d/%d)",
-			buf, len, map_len);
+	if (len != map_len) {
 		close(fd);
-		exit(1);
+		bail("failed to write to %s: (%d/%d)", path, len, map_len);
 	}
 
 	close(fd);
 }
 
 static void update_process_uidmap(int pid, char *map, int map_len)
 {
-	if ((map == NULL) || (map_len <= 0)) {
+	if (map == NULL || map_len <= 0)
 		return;
-	}
 
 	update_process_idmap("/proc/%d/uid_map", pid, map, map_len);
 }
 
 static void update_process_gidmap(int pid, uint8_t is_setgroup, char *map, int map_len)
 {
-	if ((map == NULL) || (map_len <= 0)) {
+	if (map == NULL || map_len <= 0)
 		return;
-	}
 
-	if (is_setgroup == 1) {
-		int	fd;
-		int	len;
-		char	buf[PATH_MAX];
+	if (is_setgroup) {
+		int	fd, len;
+		char path[PATH_MAX], *allow = "allow";
 
-		len = snprintf(buf, sizeof(buf), "/proc/%d/setgroups", pid);
-		if (len < 0) {
-			pr_perror("failed to get setgroups path for %d", pid);
-			exit(1);
-		}
+		len = snprintf(path, PATH_MAX, "/proc/%d/setgroups", pid);
+		if (len < 0)
+			bail("failed to get setgroups path for %d", pid);
 
-		fd = open(buf, O_RDWR);
-		if (fd == -1) {
-			pr_perror("failed to open %s", buf);
-			exit(1);
-		}
-		if (write(fd, "allow", 5) != 5) {
-			// If the kernel is too old to support
-			// /proc/PID/setgroups, write will return
-			// ENOENT; this is OK.
+		/* XXX: What about old kernels. */
+		fd = open(path, O_RDWR);
+		if (fd < 0)
+			bail("failed to open %s", path);
+
+		if (write(fd, allow, strlen(allow)) != strlen(allow)) {
```

Comment:
is it just me who prefer `write(fd, "allow", 5) != 5` :p, its  simpler ( to me ) that way :D ?